pinc/page_controls.inc

```diff
@@ -103,12 +22,15 @@ function get_proofreading_interface_data_js()
         "showText" => _("Show Text only"),
         "showImageText" => _("Show Image & Text"),
         "selectProject" => _("Select Project"),
-        "optional" => _("(optional)"),
         "reset" => _("Select a different project"),
         "projectid" => _("Project ID"),
         "selectAPage" => _("Please select a page"),
         "browserDefault" => BROWSER_DEFAULT_STR,
-        "displayImageX" => _("Display Image: %s"),
+        "browsePages" => _("Browse pages"),
+        "displayPageX" => _("Display Page: %s"),
+        "noPages" => _("There are no pages in this project"),
+        "enterID" => _("Please enter a project ID"),
+        "noPageX" => _("There is no page %s in this project"),
```

Comment:
In testing, I ran into this as "There is no page undefined in this project", which didn't make sense to me. Looking at the code, it looks like `%s` is supposed to be an image filename, but for some reason, the image filename is not being picked up when I click on the link, so it comes through as the very confusing "undefined"?

Reply:
> In my testing just now, though, while the page that I brought up for context from "Candidates for Good Words List from WordCheck" page works fine bringing up the context image, the corresponding operation from the link brought up from "Candidates for Good Words List from Proofreaders" did not. For example, [this page](https://www.pgdp.org/~rp31/c.branch/browse_api_2/tools/project_manager/show_good_word_suggestions_detail.php?projectid=projectID5e23800f9f15c&word=666f7265746f70) shows "Please select a page" in the pull-down in the image pane, and when I click on the link in the list to the left, I get a message that says "There is no page undefined in this project".

I can't reproduce these problems. Can you have a look at the browser console to see if it shows any error messages?

Reply:
I saw the "There is no page undefined in this project" message when first starting up my windows computer. I think it was on a tab that was already open in the browser. On refreshing the page it went away so it looks like a transient caching problem.